cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/getresourcerefs_test.go

```diff
@@ -21,230 +21,67 @@ import (
 	"github.com/google/go-cmp/cmp/cmpopts"
 	"github.com/kubeapps/kubeapps/cmd/apprepository-controller/pkg/apis/apprepository/v1alpha1"
 	corev1 "github.com/kubeapps/kubeapps/cmd/kubeapps-apis/gen/core/packages/v1alpha1"
-	"github.com/kubeapps/kubeapps/cmd/kubeapps-apis/plugins/pkg/resourcerefs"
+	"github.com/kubeapps/kubeapps/cmd/kubeapps-apis/plugins/pkg/resourcerefs/resourcerefstest"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type TestCase struct {
-	name               string
-	existingReleases   []resourcerefs.TestReleaseStub
-	request            *corev1.GetInstalledPackageResourceRefsRequest
-	expectedResponse   *corev1.GetInstalledPackageResourceRefsResponse
-	expectedStatusCode codes.Code
-}
-
 func TestGetInstalledPackageResourceRefs(t *testing.T) {
 
-	// TODO (gfichtenholt) what's missing here is a call to
-	// resourcerefs_test.go init() fuction. I spent quite some time but have not yet
-	// figured out how to make that happen. So I am commenting this test out until I do
+	// sanity check
+	if len(resourcerefstest.TestCases1) < 11 {
+		t.Fatalf("Expected array [resourcerefstest.TestCases1] size of at least 11")
+	}
 
-	if len(resourcerefs.TestCases1) == 0 {
-		t.Logf("Expected non-empty array [resourcerefs.TestCases1]")
+	type testCase struct {
+		baseTestCase       resourcerefstest.TestCase
+		request            *corev1.GetInstalledPackageResourceRefsRequest
+		expectedResponse   *corev1.GetInstalledPackageResourceRefsResponse
+		expectedStatusCode codes.Code
 	}
 
-	testCases := []TestCase{
-		/*
-				{
-					name:             resourcerefs.TestCases1[0].Name,
-					existingReleases: resourcerefs.TestCases1[0].ExistingReleases,
-					request: &corev1.GetInstalledPackageResourceRefsRequest{
-						InstalledPackageRef: &corev1.InstalledPackageReference{
-							Context: &corev1.Context{
-								Cluster:   "default",
-								Namespace: "default",
-							},
-							Identifier: "my-apache",
-						},
-					},
-					expectedResponse: &corev1.GetInstalledPackageResourceRefsResponse{
-						Context: &corev1.Context{
-							Cluster:   "default",
-							Namespace: "default",
-						},
-						ResourceRefs: resourcerefs.TestCases1[0].ExpectedResourceRefs,
+	newTestCase := func(tc int, identifier string, response bool, code codes.Code) testCase {
```

Comment:
yeah, that took a while to get working. My idea was that having moved some of the functionality to pkg (i.e. ResourceRefsFromManifest call), I generally want to have unit test cases there that test that func is working as expected. So I took the unit tests you had written for GetInstalledPackageResourceRefs and re-wrote them so they only test the ResourceRefsFromManifest part.  Now, that's wondeful, but the function itself is not used in isolation, there is a higher level module (helm plugin) that has an API (GetInstalledPackageResourceRefs) which calls ResourceRefsFromManifest() *AMONG* other things it does. There are a few other funcs being called in the scope of  GetInstalledPackageResourceRefs. 
A clumsy analogy to help see thing better: 
- ResourceRefsFromManifest is a unit test scenario
- GetInstalledPackageResourceRefs is an integration test scenario

If I had removed the unit tests from this module, we'd loose the "integration test" functionality. So I did not want to lose the hard work that went into writing that unit test, and instead just re-used some of the data from the "base" test as I call it. Basically, I didn't want to loose any of your hard work. This isn't set in stone. I am willing to reconsider this if you feel strongly about it. Let me know

I will add a comment to that effect in the code to hopefully make things a little clearer

Reply:
Yep, I think that's fine for now. I'll take a look next time I'm needing to touch the tests... I'm all for not removing them and re-using setup code, just keen to avoid running the same tests, but it may not be possible. Will see. Thanks!